Source/WebCore/platform/EventTrackingRegions.cpp

```diff
@@ -26,50 +26,91 @@
 #include "config.h"
 #include "EventTrackingRegions.h"
 
+#include "EventNames.h"
+
 namespace WebCore {
 
-ASCIILiteral EventTrackingRegions::eventName(Event event)
+ASCIILiteral EventTrackingRegions::eventName(EventType eventType)
 {
-    switch (event) {
-    case Event::Mousedown:
+    switch (eventType) {
+    case EventType::Mousedown:
         return "mousedown"_s;
-    case Event::Mousemove:
+    case EventType::Mousemove:
         return "mousemove"_s;
-    case Event::Mouseup:
+    case EventType::Mouseup:
         return "mouseup"_s;
-    case Event::Mousewheel:
+    case EventType::Mousewheel:
         return "mousewheel"_s;
-    case Event::Pointerdown:
+    case EventType::Pointerdown:
         return "pointerdown"_s;
-    case Event::Pointerenter:
+    case EventType::Pointerenter:
         return "pointerenter"_s;
-    case Event::Pointerleave:
+    case EventType::Pointerleave:
         return "pointerleave"_s;
-    case Event::Pointermove:
+    case EventType::Pointermove:
         return "pointermove"_s;
-    case Event::Pointerout:
+    case EventType::Pointerout:
         return "pointerout"_s;
-    case Event::Pointerover:
+    case EventType::Pointerover:
         return "pointerover"_s;
-    case Event::Pointerup:
+    case EventType::Pointerup:
         return "pointerup"_s;
-    case Event::Touchend:
+    case EventType::Touchend:
         return "touchend"_s;
-    case Event::Touchforcechange:
+    case EventType::Touchforcechange:
         return "touchforcechange"_s;
-    case Event::Touchmove:
+    case EventType::Touchmove:
         return "touchmove"_s;
-    case Event::Touchstart:
+    case EventType::Touchstart:
         return "touchstart"_s;
-    case Event::Wheel:
+    case EventType::Wheel:
         return "wheel"_s;
     }
     return ASCIILiteral();
 }
 
-TrackingType EventTrackingRegions::trackingTypeForPoint(Event event, const IntPoint& point)
+const AtomString& EventTrackingRegions::eventNameAtomString(const EventNames& eventNames, EventType eventType)
```

Comment:
Glad to see us taking advantage of the pre-existing atom strings. I’m not sure we also need the ASCII literal version of `eventName`. Is there a case where the greater efficiency of having a literal is critical?

Reply:
Yeah :D `eventNames` is materialized in the main thread and worker threads. But we can avoid materializing it in the scrolling thread, and we can save 260~ AtomString allocations in the scrolling thread only for some logging features :)